src/arkode/arkode_lsrkstep_io.c

```diff
@@ -0,0 +1,755 @@
+  ---------------------------------------------------------------*/
+int LSRKStepSetDomEigFn(void* arkode_mem, ARKDomEigFn dom_eig)
+{
+  ARKodeMem ark_mem;
+  ARKodeLSRKStepMem step_mem;
+  int retval;
+
+  /* access ARKodeMem and ARKodeLSRKStepMem structures */
+  retval = lsrkStep_AccessARKODEStepMem(arkode_mem, __func__, &ark_mem,
+                                        &step_mem);
+  if (retval != ARK_SUCCESS) { return retval; }
+
+  /* set the dom_eig routine pointer, and update relevant flags */
+  if (dom_eig != NULL)
+  {
+    step_mem->dom_eig_fn = dom_eig;
+
+    return ARK_SUCCESS;
+  }
+  else
+  {
+    step_mem->dom_eig_fn = NULL;
+
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "Internal dom_eig is not supported yet!");
+    return ARK_ILL_INPUT;
+  }
+}
+
+/*---------------------------------------------------------------
+  LSRKStepSetDomEigFrequency sets dom_eig computation frequency -
+  Dominated Eigenvalue is recomputed after "nsteps" successful steps.
+
+  nsteps = 0 refers to constant dominant eigenvalue
+  nsteps < 0 resets the default value 25 and sets nonconstant dominant eigenvalue
+  ---------------------------------------------------------------*/
+int LSRKStepSetDomEigFrequency(void* arkode_mem, int nsteps)
+{
+  ARKodeMem ark_mem;
+  ARKodeLSRKStepMem step_mem;
+  int retval;
+
+  /* access ARKodeMem and ARKodeLSRKStepMem structures */
+  retval = lsrkStep_AccessARKODEStepMem(arkode_mem, __func__, &ark_mem,
+                                        &step_mem);
+  if (retval != ARK_SUCCESS) { return retval; }
+
+  if (nsteps < 0)
+  {
+    step_mem->dom_eig_freq = DOM_EIG_FREQ_DEFAULT;
+    step_mem->const_Jac    = SUNFALSE;
+  }
+
+  if (nsteps == 0)
+  {
+    step_mem->const_Jac    = SUNTRUE;
+    step_mem->dom_eig_freq = 1;
+  }
+  else
+  {
+    step_mem->dom_eig_freq = nsteps;
+    step_mem->const_Jac    = SUNFALSE;
+  }
+
+  return ARK_SUCCESS;
+}
+
+/*---------------------------------------------------------------
+  LSRKStepSetMaxNumStages sets the maximum number of stages allowed.
+  If the combination of the maximum number of stages and the current 
+  time step size in the LSRKStep module does not allow for a stable 
+  step, the step routine returns to ARKODE for an updated (refined) 
+  step size. The number of such returns is tracked in a counter, 
+  which can be accessed using ARKodeGetNumExpSteps.
+  ---------------------------------------------------------------*/
+int LSRKStepSetMaxNumStages(void* arkode_mem, int stage_max_limit)
+{
+  ARKodeMem ark_mem;
+  ARKodeLSRKStepMem step_mem;
+  int retval;
+
+  /* access ARKodeMem and ARKodeLSRKStepMem structures */
+  retval = lsrkStep_AccessARKODEStepMem(arkode_mem, __func__, &ark_mem,
+                                        &step_mem);
+  if (retval != ARK_SUCCESS) { return retval; }
+
+  if (stage_max_limit < 2 || stage_max_limit > 10000)
```

Comment:
If we do keep an upper bound
```suggestion
  if (stage_max_limit < 2 || stage_max_limit > STAGE_MAX_LIMIT_DEFAULT)
```

Reply:
We have changed this number a lot, so I totally agree your argument. I think we should have two default values: one for the default value of the stage_max_limit = 200 and the other for the max allowable (by a user set) such numbers, 10000.